tests/test_optimize.py

```diff
@@ -54,25 +54,25 @@
 univariate_funcs = [np.sin,
                     lambda x: np.exp(x / 10.),
                     lambda x: x ** 2]
-grad_uni_fns = [np.cos,
-                lambda x: np.exp(x / 10.) / 10.,
-                lambda x: 2 * x]
+grad_uni_fns = [lambda x: (np.cos(x), ),
```

Comment:
Maybe it's better to make these more explicit by passing a single element list to `tuple`, just because using the `(x,)` syntax for creating single element tuples could be missed. Would just help perhaps for future reference with readability.
```suggestion
grad_uni_fns = [lambda x: tuple([np.cos(x)]),
```

Reply:
good point. that does look a lot clearer